src/main/java/com/pinterest/secor/parser/PartitionFinalizer.java

```diff
@@ -168,7 +168,7 @@ private void finalizePartitionsUpTo(String topic, String[] uptoPartitions) throw
                             LOG.warn("HivePrefix is not defined.  Skip hive registration");
                         }
                     }
-                    if (hiveTableName != null) {
+                    if (hiveTableName != null && !mConfig.getSkipQuboleAddPartition()) {
```

Comment:
double negate might be a bit confusing.  It might be better to have a positive parameter name, e.g. enableQubole = true